### ISSUE TITLE
Prevent 'Undefined array key "title"' PHP warning in Checkout block

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#46238]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In p1712238888027759-slack-C011ENB20Q1, GS reported seeing the PHP warning `Undefined array key "title"` on the Checkout block. This PHP warning is caused by the following section:

https://github.com/woocommerce/woocommerce/blob/51dfbb969f397692a6065288e2947f8affcd56b6/plugins/woocommerce/src/Blocks/BlockTypes/Checkout.php#L353-L355

We do register the option `woocommerce_pickup_location_settings` here:

https://github.com/woocommerce/woocommerce/blob/51dfbb969f397692a6065288e2947f8affcd56b6/plugins/woocommerce/src/Blocks/Shipping/ShippingController.php#L167-L201

But we do not initialise the values. As long as a merchant hasn't enabled the Local Pickup option via `wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`, the option `woocommerce_pickup_location_settings` returns an empty array. 

### Expected behavior

No PHP warning should be thrown.

### Actual behavior

A PHP warning will be thrown.

### Steps to reproduce

1. Connect to the database of your site and delete `woocommerce_pickup_location_settings` from `wp_options`.
2. Install and activate the [Query Monitor](https://wordpress.org/plugins/query-monitor/) plugin.
3. Go to the frontend.
4. Add a product to the cart.
5. Go to the checkout page (using the Checkout block).
6. See the PHP warning `Undefined array key "title"` in the Query Monitor.

<img width="1352" alt="Screenshot 2024-04-04 at 22 58 09" src="https://github.com/woocommerce/woocommerce/assets/3323310/bc2bd707-9259-4d3d-9650-07ef5bcf0aad">

### WordPress Environment

<details>
<summary>WordPress Environment</summary>

```
### WordPress Environment ###

WordPress address (URL): https://store.test
Site address (URL): https://store.test
WC Version: 8.9.0
REST API Version: ✔ 8.9.0
Action Scheduler Version: ✔ 3.7.3
Log Directory Writable: ✔
WP Version: 6.5
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.4
PHP Version: 8.1.27
PHP Post Max Size: 512 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.6.0
(SecureTransport) OpenSSL/3.2.1

SUHOSIN Installed: –
MySQL Version: 8.3.0
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.9.0
WC Database Prefix: wp_
Total Database Size: 6.67MB
Database Data Size: 4.65MB
Database Index Size: 2.02MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.11MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 3.05MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.13MB + Index: 0.11MB + Engine InnoDB
wp_posts: Data: 0.16MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_indexable: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 22
customize_changeset: 1
page: 11
post: 16
product: 19
product_variation: 10
revision: 27
shop_coupon: 1
shop_order_placehold: 8
wp_global_styles: 3
wp_navigation: 1
wp_template_part: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

Query Monitor: by John Blackbourn – 3.15.0
Code Snippets: by Code Snippets Pro – 3.6.4
WooCommerce: by Automattic – 8.9.0-dev

### Inactive Plugins (2) ###

Simple Custom CSS and JS: by SilkyPress.com – 3.46
WooCommerce Stripe Gateway: by WooCommerce – 8.1.0

### Dropin Plugins () ###

db.php: Query Monitor Database Class (Drop-in)

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 119 KB

### WC Pages ###

Shop base: #54 - /products-beta/classic-shop/
Cart: #56 - /cart/
Checkout: #57 - /checkout/
My account: #60 - /my-account/
Terms and conditions: #61 - /terms/

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/error.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/notice.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
product-pre-publish-modal
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-custom-fields
settings
async-product-editor-category-field
launch-your-store

Daily Cron: ✔ Next scheduled: 2024-04-05 06:00:36 +00:00
Options: ✔
Notes: 75
Onboarding: -

### Action Scheduler ###

Complete: 238
Oldest: 2024-03-19 05:58:20 +0000
Newest: 2024-04-04 15:28:06 +0000

Failed: 2
Oldest: 2024-03-27 17:33:01 +0000
Newest: 2024-03-28 13:48:08 +0000

Pending: 3
Oldest: 2024-04-04 17:50:33 +0000
Newest: 2024-04-05 01:43:55 +0000


### Status report information ###

Generated at: 2024-04-04 15:59:02 +00:00
```

</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by opr** [comment]
Thanks for reporting this Niels, this was fixed with https://github.com/woocommerce/woocommerce/pull/46182

**Comment by nielslange** [comment]
> Thanks for reporting this Niels, this was fixed with #46182

Thanks for letting me know that this issue was fixed with your PR, @opr. 🙌
